pages/_app.js

```diff
@@ -50,6 +50,32 @@ export default function App({ Component, pageProps }) {
     ]);
   }
 
+  function handleProjectDetailsChange(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    setProjects(
+      projects.map((entry) =>
+        entry.id === currentProject.id ? { ...entry, ...data } : entry
+      )
+    );
+  }
+
+  function handleViewDetailsChange(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    setViews(
+      views.map((entry) =>
+        entry.id === currentView.id ? { ...entry, ...data } : entry
+      )
```

Comment:
I have a question. Is a view independent from a project or is it strictly tied to it. If the latter one is true you have two options:
1. the views need to moved into the data structure of your project objects. That means that there should be a new key value pair in each project object that is called "views". The value is an array of your view objects that is linked to that project.
2. You create an independent views list like you did here. but each project still needs an "views" key which is a list of __view IDs__. If updating your views you need to update the views list as well as the respective project (if you add or remove a view from the project). 

It is your choice which way to go, I guess the first one is easier at the start.

Reply:
They are independet from each other 👍